### PR TITLE
Fe 248 generic table shared components

### PR DESCRIPTION
**Pull request by Cherna**
Developer considerations
--------------

- [ ] Increment the version number for relevant packages
  - [ ] This is a new MAJOR version (incompatible API changes were made)
  - [ ] This is a new MINOR version (functionality was added in a backwards compatible manner)
  - [ ] This is a new PATCH version (backwards compatible bug fixes)
- [ ] Update the CHANGELOG.md with a summary of the changes
  - If this is a MAJOR version, what are the breaking API changes? Was anything added to the API?
  - If this is a MINOR version, what feature was added? Why would consumers want this new version?
  - If this is a PATCH version, what was fixed?
- [ ] Update the appropriate documentation. Keep in mind that this mono repository has README files at the repository, project, and component levels. We want developers to be able to find what they need from the main README, so make sure to add links to component documentation if adding something new.

### CODE REVIEW

File: stencil-library/src/components/table/table.tsx [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        <div part="loading-state-spinner" class="spinner-border" role="status">
+          <span class="visually-hidden">Loading...</span>
+        </div>
+      </td>
+    </tr>
+  );
+
+  paginationBar = () => {
+    return (
+      <div class="pagination-bar d-flex justify-content-center gap-3">
```
Comment (jakemer) [inline stencil-library/src/components/table/table.tsx:66]: Are `pagination-bar`, `pagination-btn`, `pagination-prev-btn` etc being used? I don't see the styles anywhere

File: stencil-library/src/components/table/table.tsx [hunk 2]
```diff
@@ -0,0 +1,138 @@
+import { Component, Host, h, Prop } from '@stencil/core';
+import { PagingInfo, pagingDefaults } from '../table/table-utils';
+
+interface ExtendedPagingInfo extends PagingInfo {
+  onPrev: () => void;
+  onNext: () => void;
+}
+
+const ExtendedPagingDefaults = {
```
Comment (jakemer) [inline stencil-library/src/components/table/table.tsx:9]: Should this be `const ExtendedPagingDefaults: ExtendedPagingInfo =`?

Comment (Cherna) [reply stencil-library/src/components/table/table.tsx:9]: Yes! Good catch.

File: stencil-library/src/components/table/table.tsx (hunk 1 above)
Comment (Cherna) [reply stencil-library/src/components/table/table.tsx:66]: Nope, can be removed safely.

**Review by jakemer — APPROVED** [review]

**Pull request merged**